packages/docusaurus/src/server/plugins/init.ts

```diff
@@ -49,9 +76,27 @@ export default function initPlugins({
       const pluginModule: any = importFresh(
         pluginRequire.resolve(pluginModuleImport),
       );
-      return (pluginModule.default || pluginModule)(context, pluginOptions);
+
+      const plugin = pluginModule.default || pluginModule;
+      if (plugin.validateOptions) {
+        const options = plugin.validateOptions({
+          validate,
+          options: pluginOptions,
+        });
+        pluginOptions = options;
+      }
+      if (plugin.validateThemeConfig) {
+        const validatedTheme = plugin.validateThemeConfig({
+          validate: validateAndStrip,
+          themeConfig: context.siteConfig.themeConfig,
+        });
+        context.siteConfig.themeConfig = {
+          ...context.siteConfig.themeConfig,
+          ...validatedTheme,
```

Comment:
Why do you merge here?

Do you want to support the usecase where the user validation does not return a new object? (like validation but no normalization?)

In such case I think we should do the same for plugins as well, and make it more explicit.

like 

```js
if ( typeof validatedTheme !== "undefined" ) {
  return validatedTheme;
}
else {
  return context.siteConfig.themeConfig;
}
```

In any case, I'm not fan of enabling the user to return "partiel theme config" from the `validateThemeConfig` fn

Reply:
Now u have pointed it out, I am also not a big fan of this method but I was trying to support theme which does not validate their options. I was thinking to use it to force plugin author to write validation.

Reply:
It make more sense to remove it and not normalize themeConfig.

Reply:
I removed normalizing themeConfig